cloverage/src/cloverage/coverage.clj

```diff
@@ -160,16 +162,17 @@
   (binding [*ns* (find-ns 'clojure.core)]
     (eval `(dosync (alter clojure.core/*loaded-libs* conj '~namespace)))))
 
-(defn find-nses [ns-path regex-patterns]
-  "Given ns-path and regex-patterns returns:
-  * empty sequence when ns-path is nil and regex-patterns is empty
-  * all namespaces on ns-path (if regex-patterns is empty)
-  * all namespaces on the classpath that match any of the regex-patterns (if ns-path is nil)
-  * namespaces on ns-path that match any of the regex-patterns"
+(defn find-nses [ns-paths regex-patterns]
+  "Given ns-paths and regex-patterns returns:
+  * empty sequence when ns-paths is empty and regex-patterns is empty
+  * all namespaces on all ns-paths (if regex-patterns is empty)
+  * all namespaces on the classpath that match any of the regex-patterns (if ns-paths is empty)
+  * namespaces on ns-paths that match any of the regex-patterns"
   (let [namespaces (->> (cond
-                          (and (nil? ns-path) (empty? regex-patterns)) '()
-                          (nil? ns-path) (blt/namespaces-on-classpath)
-                          :else (blt/namespaces-on-classpath :classpath ns-path))
+                          (and (empty? ns-paths) (empty? regex-patterns)) '()
+                          (empty? ns-paths) (blt/namespaces-on-classpath)
+                          :else (mapcat #(blt/namespaces-on-classpath :classpath %)
+                                        ns-paths))
```

Comment:
I'm not sure a threading macro is appropriate here, but you didn't introduce it so I guess that's a moot point. Still interested in your opinion though.

Reply:
Yeah, in my opinion it tends to obscure what's happening. Especially since that first `cond` is handling so many corner cases.

Reply:
lein-kibit doesn't like it either.

```
$ lein kibit
At src/cloverage/coverage.clj:169:
Consider using:
  (map
    name
    (cond
      (and (nil? ns-path) (empty? regex-patterns)) '()
      (nil? ns-path) (blt/namespaces-on-classpath)
      :else (blt/namespaces-on-classpath :classpath ns-path)))
instead of:
  (->>
    (cond
      (and (nil? ns-path) (empty? regex-patterns)) '()
      (nil? ns-path) (blt/namespaces-on-classpath)
      :else (blt/namespaces-on-classpath :classpath ns-path))
    (map name))
```

Reply:
I'd probably move the `cond` into the `let`.

Reply:
Let's hold off on this -- I'm implementing kibit and eastwood on a separate branch (you have another ticket open for that), and I'm going to deal with all the changes there.  It's going to be a lot of little tweaks everywhere.

Reply:
When you say "hold off on this": you don't mean "this PR", but rather, "fixing style issues", right? If so: agreed.

Reply:
Yes, agreed, I meant hold off on the style issues & instead do that in the other PR..